tests/unit/components/playback/controls/PlateBarcode.spec.js

```diff
@@ -56,102 +57,50 @@ describe("PlateBarcode.vue", () => {
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 11] value such as unicode charcter in Chinese 浩 is entered  fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA20011浩211"); // having special unicode chacters in the last
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 10] values such as unicode charcter in Chinese numbers 二千万一千〇九 (20001009) is entered  fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA二千万一千〇九"); // having special unicode chacters of chinexe numbers 二千万一千〇九 aka 20001009
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
+  test.each([
+    [
+      "AB200440012",
+      "validate_plate_barcode",
+      null, //  vuex value
+    ],
+    ["12200440012", "validate_plate_barcode", null],
+    ["*#200440012", "validate_plate_barcode", null],
+    ["MA200000012", "validate_plate_barcode", null],
+    ["MA203670012", "validate_plate_barcode", null],
+    ["MA209990121", "validate_plate_barcode", null],
+    ["MA 13000012", "validate_plate_barcode", null],
+    ["MA  300000", "validate_plate_barcode", null],
+    ["MB190440991", "validate_plate_barcode", "MB190440991"], // year 19 now allowed.
+    ["MB210440991", "validate_plate_barcode", "MB210440991"], // year 21 now allowed
+    ["MB100440991", "validate_plate_barcode", "MB100440991"], // year 10 now allowed.
+    ["MA*#300001", "validate_plate_barcode", null],
+    ["MA20222111*", "validate_plate_barcode", null],
+    ["MA20010*#12", "validate_plate_barcode", null],
+    ["MA20001 021", "validate_plate_barcode", null],
+    ["MA20001º21", "validate_plate_barcode", null],
+    ["MA20210न21", "validate_plate_barcode", null],
+    ["MA20011浩211", "validate_plate_barcode", null],
+    ["MA二千万一千〇九", "validate_plate_barcode", null],
+    ["MA", "validate_plate_barcode", null],
+    ["MA20", "validate_plate_barcode", null],
+    ["MA20044", "validate_plate_barcode", null],
+    ["MA20**#*", "validate_plate_barcode", null],
+    ["MA20044001", "validate_plate_barcode", "MA20044001"],
+    ["M120044099", "validate_plate_barcode", null], // M1 is disallow
+    ["MD20044099", "validate_plate_barcode", "MD20044099"], // new rule allow MD
+  ])(
+    "Given a barcode with text  %s, When validation rule %s criteria FAILS for invalid barcode and PASSES for valid barcode, Then only valid barcode %s is stored in Vuex playback.barcode",
+    async (platecode, validation_rule, store_data) => {
+      const spied_text_validator = jest.spyOn(
```

Comment:
you need to assert that the spied function was used---and also include in your name of the test that you are confirming that the correct text validation function was used

Reply:
New Assertion added on spied function.
`expect(spied_text_validator).toHaveBeenCalledWith(platecode);`